src/scripts/destinyTrackerApi/reviewDataCache.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'underscore';
+import { itemTransformer } from './itemTransformer.js';
+
+class reviewDataCache {
+  constructor() {
+    this._itemTransformer = new itemTransformer();
+    this._itemStores = [];
+  }
+
+  getMatchingItem(item) {
+    var dtrItem = this._itemTransformer.translateToDtrWeapon(item);
+
+    dtrItem.referenceId = String(dtrItem.referenceId);
+
+    return _.findWhere(this._itemStores, { referenceId: dtrItem.referenceId, roll: dtrItem.roll });
+  }
+
+  getRatingData(item) {
+    var matchingItem = this.getMatchingItem(item);
+
+    if (!matchingItem) {
+      return null;
+    }
+
+    return matchingItem;
+  }
+
+  toAtMostOneDecimal(rating) {
+    if (!rating) {
+      return null;
+    }
+
+    if ((rating % 1) === 0) {
+      return rating;
+    }
+
+    return rating.toFixed(1);
+  }
+
+  addScore(dtrRating) {
+    dtrRating.rating = this.toAtMostOneDecimal(dtrRating.rating);
+
+    this._itemStores.push(dtrRating);
+  }
+
+  addUserReviewData(item,
+                    userReview) {
+    var matchingItem = this.getMatchingItem(item);
+
+    var rating = matchingItem.rating;
+
+    Object.assign(matchingItem,
+                  userReview);
+
+    matchingItem.userRating = matchingItem.rating;
+
+    matchingItem.rating = rating;
+  }
+
+  addReviewsData(item,
+                 reviewsData) {
+    var matchingItem = this.getMatchingItem(item);
+
+    matchingItem.reviews = reviewsData.reviews;
+    matchingItem.reviewsDataFetched = true;
+  }
+
+  getItemStores() {
+    return this._itemStores;
+  }
+
+  eventuallyPurgeCachedData(item) {
+    var tenMinutes = 1000 * 60 * 10;
+    var self = this;
+
+    setTimeout(function() {
+      var matchingItem = self.getMatchingItem(item);
+
+      matchingItem.reviews = null;
+      matchingItem.reviewsDataFetched = false;
+    },
+      tenMinutes);
```

Comment:
Yeah, I'm suggesting that instead of scheduling a task to null out the info, just keep track of the expiration date and don't return the info if it's out of date. See `dimVendorsService` as an example of that sort of caching.